joomla/joomla-cms
# Make sure a value is passed to base64_decode

### Summary of Changes
When installing a module on PHP 8.1 the following notice is observed in the Network Inspector:
`base64_decode(): Passing null to parameter #1 ($string) of type string is deprecated`

When the base64 issue is solved you get another notice:
`strpos(): Passing null to parameter #1 ($haystack) of type string is deprecated`

This happens because there is no `return` value in the POST request. Not having the `return` value is absolutely fine but we need to check if it has a value.

This PR fixes that.

### Testing Instructions
1. Make sure you are running PHP 8.1
2. In the Global Settings set the Error Reporting to `maximum`
3. Go to System -> Install -> Extensions
4. Open the Developer Tools of the browser
5. Click on the Network tab
6. Install the attached module
[mod_test.zip](https://github.com/joomla/joomla-cms/files/7874659/mod_test.zip)
5. You should see this error
![image](https://user-images.githubusercontent.com/359377/149616785-fce75ee5-a1ab-4795-a38d-4cb61cb60817.png)
6. In the Network tab click on the request that just happened
7. Click on the Response tab
8. You will see the notice there. If you do not see the notice, you may need to toggle the Raw button
9. Apply the patch
10. Install the module again
11. Click on the Response tab
12. You will see no notice here


### Actual result BEFORE applying this Pull Request
An error is shown due to 8.1 compatibility


### Expected result AFTER applying this Pull Request
The 8.1 compatibility is solved.


### Documentation Changes Required
None
